plugin/src/main/java/com/denizenscript/denizen/events/item/ItemRecipeFormedScriptEvent.java

```diff
@@ -88,15 +85,19 @@ public ObjectTag getContext(String name) {
                         recipe.addObject(new ItemTag(Material.AIR));
                     }
                 }
-                return recipe;
+                yield recipe;
             }
-            case "recipe_id":
+            case "recipe_id" -> {
                 if (event.getRecipe() instanceof Keyed) {
-                    return new ElementTag(((Keyed) event.getRecipe()).getKey().toString());
+                    yield new ElementTag(((Keyed) event.getRecipe()).getKey().toString());
                 }
-                break;
-        }
-        return super.getContext(name);
+                else {
+                    yield super.getContext(name);
+                }
+            }

```

Comment:
Can just return `null` if the context was used but the relevant values aren't valid - `super.getContext` does stuff like the `cancelled` context, which obviously won't be valid here.
This can probably be in-lined using pattern matching as well, e.g. `-> event.getRecipe() instanceof Keyed keyed ? new ElementTag(keyed, true) : null;`